controllers/hostedcontrolplane/hostedcontrolplane.go

```diff
@@ -323,3 +358,145 @@ func (r *HostedControlPlaneReconciler) SetupWithManager(mgr ctrl.Manager) error
+	}
+	valueDynatraceApiToken := string(valueBytesDynatraceApiToken)
+
+	valueBytesDynatraceTenant, ok := secret.Data[dynatraceTenantKey]
+	if !ok {
+		return "", "", fmt.Errorf("secret did not contain key %s", dynatraceTenantKey)
+	}
+	if len(valueBytesDynatraceTenant) == 0 {
+		return "", "", fmt.Errorf("%s is empty", dynatraceTenantKey)
+	}
+	valueDynatraceTenant := string(valueBytesDynatraceTenant)
+
+	return valueDynatraceApiToken, valueDynatraceTenant, nil
+}
+
+func getDynatraceHttpMonitorId(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, bool) {
+	labels := hostedcontrolplane.GetLabels()
+	dynatraceHttpMonitorId, ok := labels[httpMonitorLabel]
+	return dynatraceHttpMonitorId, ok
+}
+
+func (r *HostedControlPlaneReconciler) UpdateHostedControlPlaneLabels(ctx context.Context, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane, key, value string) error {
+	labels := hostedcontrolplane.GetLabels()
+	labels[key] = value
+	hostedcontrolplane.SetLabels(labels)
+
+	err := r.Client.Update(ctx, hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("error updating hostedcontrolplane monitor: %v", err)
+	}
+	return nil
+}
+
+func GetAPIServerHostname(hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (string, error) {
+	for _, service := range hostedcontrolplane.Spec.Services {
+		if service.Service == "APIServer" {
+			return service.ServicePublishingStrategy.Route.Hostname, nil
+		}
+	}
+	return "", fmt.Errorf("APIServer service not found in the hostedcontrolplane")
+}
+
+func (r *HostedControlPlaneReconciler) deployDynatraceHttpMonitorResources(ctx context.Context, dynatraceApiClient *dynatrace.DynatraceApiClient, log logr.Logger, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) error {
+	//if http monitor does not exist, and hcp is not marked for deletion, and hcp is ready, then create http monitor
+	//get apiserver
+	apiServerHostname, err := GetAPIServerHostname(hostedcontrolplane)
+	if err != nil {
+		return fmt.Errorf("failed to get APIServer hostname %v", err)
+	}
+	monitorName := strings.Replace(apiServerHostname, "api.", "", 1)
+	// apiServerHostname := hostedcontrolplane.Spec.Services[1].ServicePublishingStrategy.Route.Hostname
+	monitorLocation := hostedcontrolplane.Spec.Platform.AWS.EndpointAccess
+
+	//in hcp, spec.services.service["APIServer"].servicePublishingStrategy.route.hostname is api.test-rs1.dgcj.i3.devshift.org
+	// apiUrl := "https://api.hb-testing.j1b6.i3.devshift.org/livez"
+
+	apiUrl := fmt.Sprintf("https://%s/livez", apiServerHostname)
+
+	dynatraceHttpMonitorId, ok := getDynatraceHttpMonitorId(hostedcontrolplane)
+	if ok {
+		exists, err := dynatraceApiClient.ExistsHttpMonitorInDynatrace(dynatraceHttpMonitorId)
+		if err != nil {
+			return fmt.Errorf("failed calling ExistsHttpMonitorInDynatrace: %v", err)
+		}
+		if exists {
+			log.Info(fmt.Sprintf("HTTP monitor label found. Skipping creating a monitor. Monitor id is %s", dynatraceHttpMonitorId))
+			return nil
+		}
+	}
```

Comment:
```suggestion
	dynatraceHttpMonitorId, ok := getDynatraceHttpMonitorId(hostedcontrolplane)
	if ok {
		exists, err := dynatraceApiClient.ExistsHttpMonitorInDynatrace(dynatraceHttpMonitorId)
		if err != nil {
			return fmt.Errorf("failed calling ExistsHttpMonitorInDynatrace: %v", err)
		}
		if exists {
			log.Info(fmt.Sprintf("HTTP monitor label found. Skipping creating a monitor. Monitor id is %s", dynatraceHttpMonitorId))
			return nil
		}
	}
```

Minor nit: would it make sense to encapsulate this entire stanza as a function (ie - `checkIfMonitorExists`) for readability? The goal here is to just bail early if RMO's already done everything it needs to, right?

Reply:
Updated and added tests. Now we are calling  `checkHttpMonitorExists` function in `hostedcontrolplane.go`. 
```go
exists, err := checkHttpMonitorExists(dynatraceApiClient, hostedcontrolplane)
	if err != nil {
		return fmt.Errorf("failed to check http monitor exists %v", err)
	}
	if exists {
		log.Info(fmt.Sprintf("HTTP monitor label found. Skipping creating a monitor for %s", monitorName))
		return nil
	}
```
```go
func checkHttpMonitorExists(dynatraceApiClient *dynatrace.DynatraceApiClient, hostedcontrolplane *hypershiftv1beta1.HostedControlPlane) (bool, error) {
	dynatraceHttpMonitorId, ok := getDynatraceHttpMonitorId(hostedcontrolplane)
	if ok {
		exists, err := dynatraceApiClient.ExistsHttpMonitorInDynatrace(dynatraceHttpMonitorId)
		if err != nil {
			return false, fmt.Errorf("failed calling ExistsHttpMonitorInDynatrace [monitorId:%s]: %v", dynatraceHttpMonitorId, err)
		}
		if exists {
			return true, nil
		}
	}
	return false, nil
}
```